src/Layout/toolset-tasks/UpdatePortableRuntimeIdentifierGraph.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Task = Microsoft.Build.Utilities.Task;
+
+namespace Microsoft.DotNet.Cli.Build
+{
+    public class UpdatePortableRuntimeIdentifierGraph : Task
+    {
+        [Required]
+        public string InputFile { get; set; }
+
+        [Required]
+        public string OutputFile { get; set; }
+
+
+        //  ItemSpec should be a RID, and "Imports" metadata should be a semicolon-separated list of RIDs that the ItemSpec RID imports
+        public ITaskItem[] AdditionalRuntimeIdentifiers { get; set; }
+
+        public override bool Execute()
+        {
+            JToken json;
+
+            using (var file = File.OpenText(InputFile))
+            using (JsonTextReader reader = new JsonTextReader(file))
+            {
+                json = JObject.ReadFrom(reader);
+            }
+
+            JObject runtimes = (JObject) json["runtimes"];
+
+            if (AdditionalRuntimeIdentifiers != null)
+            {
+                foreach (var rid in AdditionalRuntimeIdentifiers)
+                {
+                    var importedRids = rid.GetMetadata("Imports").Split(';');
+                    runtimes.Add(rid.ItemSpec, new JObject(new JProperty("#import", new JArray(importedRids))));
```

Comment:
I think this is correct and updates json["runtimes"] rather than making a copy...but can we have a regression test for that? Or perhaps just add json["runtimes"] = runtimes at the end?